chronos/apps/http/httpserver.nim

```diff
@@ -905,10 +905,11 @@ proc getResponseFence*(connection: HttpConnectionRef,
     let address = connection.getRemoteAddress()
     ResponseFence.err(HttpProcessError.init(
       HttpServerError.RecoverableError, exc, address, exc.code))
-  except CatchableError as exc:
-    let address = connection.getRemoteAddress()
-    ResponseFence.err(HttpProcessError.init(
-      HttpServerError.CatchableError, exc, address, Http503))
+  except HttpResponseError as exc:
```

Comment:
as an idea, if instead we add `critical: bool` to httpresponseerror, here we can use `
```nim
  case exc.critical
  of true: ..
  of false: ..
```

derived type cannot express a finite set of options, hence we get these ugly "extra" exception handlers.

critical or not can also be an enum - if we have:
```
  enum HttpErrorKind =
     invalidParameter
     hostNotOnline
```
we can also make `const CriticalError = {HttpErrorKind.invalidParameter}` and users can write:
```
case exc.kind
of CriticalError: ...
of RecoverableError: ...
```

The downside of an enum is that if we add a new enum value, it breaks all `case` statements - as long as users use the categories / sets (`Critical` and `Recoverable`), this isn't a problem (because we can add to the categories), so maybe we can simply document that new enum values may be added in the future.

I think the field makes it easier to write good error handling code while at the same time it maintains more simple raises lists (assuming of course we're confident that we can categorise all response errors into critical or not in a meaningful way)

Reply:
hm, looking at the code, it looks like this PR already places `HttpCode` in `HttpResponseError` so all that's needed really is the critical / recoverable set .. we can even retain `HttpCriticalError` and `HttpRecoverableError` for backwards compatibility and document that they are equivalent to the `code` field - in new code, we can always catch only `HttpResponseError` so it's clean..